docs/core/eventdata/tracks.md

```diff
@@ -1,4 +1,730 @@
+track state and a track backend, where a few examples are shown, and will be
+discussed below.
+
+(edm_track_api)=
+## Track API
+
+:::{doxygenclass} Acts::TrackProxy
+:members:
+:::
+
+:::{doxygenclass} Acts::TrackContainer
+:members:
+:::
+
+(edm_track_iteration)=
+## Track state iteration and forward linking
+
+By default, track states are connected as a one-directional linked list, where
+each track state knows its *previous* track state. {numref}`ckf_tree` shows an
+example of a track state tree, like it is constructed by the combinatorial
+track finding.
+
+In {numref}`ckf_tree` states $S_7$ and $S_6$ are the two {term}`tip states<tip
+state>` of the track state tree, whil $S_1$ is the single {term}`stem state`.
+In the case of combinatorial track finding starting from e.g. a
+{term}`seed<Seed>`, it could be the location of the innermost {term}`space
+point<Space point>`.
+
+(ckf_tree)=
+:::{figure} figures/ckf_tree.svg
+:width: 400px
+:align: center
+Illustration of a branching multi-trajectory that is created during
+combinatorial track finding.
+:::
+
+Each track object points at a single {term}`tip state` to define its track state sequence.
+The {class}`Acts::TrackProxy` class has various methods to access the track state sequence:
+
+```cpp
+auto track = getTrackFromSomewhere();
+for(const auto trackState : track.trackStatesReversed()) {
+  // do something with track state
+}
+
+```
+
+Note that {func}`Acts::TrackProxy::trackStatesReversed` iterates from the {term}`tip state` to
+the {term}`stem state`, i.e. from the outside in.
+
+:::{attention}
+By-default, it is not possible to iterate *forward* through the track states on a track!
+The track's track states need to be *forward-linked* for this to be possible.
+:::
+
+The reason for this is this:
+As the trajectory branches at the second sensor into $S_2$/$S_3$, it is not
+possible to connect the states forward, i.e. store in $S_1$ what the *next*
+state is going to be: it is ambiguous!
+
+However, when track finding has concluded, and the trajectories identified by
+{term}`tip states<tip state>` $S_7$ and $S_8$ have been discarded or are being
+copied into an output container, it is possible to *forward link* the track
+state sequences. This is possible **if** the trajectory does not branch
+anymore! {func}`Acts::TrackProxy::copyFrom` will implicitly forward link the
+track states, as it is guaranteed to not branch after copying.
+
+You can manually add forward-linking to a track by calling
+{func}`Acts::TrackProxy::linkForward` or
+{func}`Acts::TrackProxy::reverseTrackStates`.
+
+:::{warning}
+Calling either {func}`Acts::TrackProxy::linkForward` or
+{func}`Acts::TrackProxy::reverseTrackStates` on a track state sequence which
+has branching will break the branching! If you have other tracks pointing at a
+{term}`tip state` that branches from the sequence you're trying to
+forward-link, it will be corrupted!
+:::
```

Comment:
should we also warn about branching + smoothing?

Reply:
Done at the end of the section.